src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageProcessor.java

```diff
@@ -76,11 +77,11 @@ public boolean processNextMessage() throws ServiceBusException, InterruptedExcep
     private MessageProcessingResult processCreateCommand(IMessage message) {
         log.info("Started processing payment message with ID {}", message.getMessageId());
 
-        CreatePaymentMessage payment = null;
+        PaymentMessage payment = null;
 
         try {
             payment = paymentMessageParser.parse(message.getMessageBody());
```

Comment:
this call returns `CreatePaymentMessage`. why this change??

Reply:
PR #72  would give an idea what  I am trying to do.
I do not want to change Processor for every new payment process.

Reply:
completely different things happen based on the type of payment message, so why not?

Reply:
anyway, this change doesn't make sense in the scope of this PR of adding super class